photomaker/model/model.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+import torch
+from huggingface_hub import hf_hub_download
+from photomaker import PhotoMakerStableDiffusionXLPipeline
+from PIL import Image
+
+
+class PhotoMakerModel:
+    def __init__(self):
+        self.model = None
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
+
+    def load_model(self):
+        # Download the model from Hugging Face Hub
+        model_path = hf_hub_download(
+            repo_id="TencentARC/PhotoMaker",
+            filename="photomaker-v1.bin",
+            repo_type="model",
+        )
+        self.model = PhotoMakerStableDiffusionXLPipeline.from_pretrained(model_path).to(
```

Comment:
i think you can `from_pretrained(model_path, device=self.device)` instead which might be faster. Can you try it?

Reply:
Also, I'm pretty sure `PhotoMakerStableDiffusionXLPipeline` is not a real class.

Reply:
The photomaker repository instantiates the pipeline this way (`PhotoMakerStableDiffusionXLPipeline` is defined in the photomaker library). I haven't tested performance in either case however.